src/Linters/PreferLambdasLinter.php

```diff
@@ -0,0 +1,98 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST\Linters;
+
+use type Facebook\HHAST\Linters\{ASTLintError, AutoFixingASTLinter};
+use type Facebook\HHAST\{
+  AmpersandToken,
+  AnonymousFunction,
+  EditableNode,
+  EqualEqualGreaterThanToken,
+  LambdaExpression,
+  LambdaSignature,
+  LeftParenToken,
+  PrefixUnaryExpression,
+  WhiteSpace,
+};
+use function Facebook\HHAST\Missing;
+use namespace HH\Lib\C;
+
+final class PreferLambdasLinter extends AutoFixingASTLinter<AnonymousFunction> {
+  <<__Override>>
+  protected static function getTargetType(): classname<AnonymousFunction> {
+    return AnonymousFunction::class;
+  }
+
+  <<__Override>>
+  protected function getTitleForFix(LintError $_): string {
+    return 'Convert to lambda';
+  }
+
+  <<__Override>>
+  public function getLintErrorForNode(
+    AnonymousFunction $node,
+    vec<EditableNode> $parents,
+  ): ?ASTLintError<AnonymousFunction> {
+
+    $use_expr = $node->getUse();
+
+    $uses_references = $use_expr is nonnull &&
+      !C\is_empty(
+        $use_expr->getDescendantsWhere(
+          ($node, $_) ==> $node is PrefixUnaryExpression &&
+            $node->getOperator() is AmpersandToken,
+        ),
+      );
+
+    if ($uses_references) {
+      return null;
+    } else {
```

Comment:
`else` unneeded because `if` body is noreturn; move else body to function body and unindent for readability